File: api/Domain/Entity/Character.cs
using Newtonsoft.Json;

namespace api.Domain.Entity
{
    public class Character
    {
        [JsonProperty("id")]
        public int Id { get; set; }

        [JsonProperty("name")]
        public string? Name { get; set; } 

        [JsonProperty("description")]
        public string? Description { get; set; }

        [JsonProperty("modified")]
        public DateTime Modified { get; set; }

        [JsonProperty("thumbnail")]
        public Thumbnail? Thumbnail { get; set; }

        [JsonProperty("resourceURI")]
        public string? ResourceURI { get; set; }

        [JsonProperty("comics")]
        public Comics? Comics { get; set; }

        [JsonProperty("series")]
        public Series? Series { get; set; }

        [JsonProperty("stories")]
        public Stories? Stories { get; set; }

        [JsonProperty("events")]
        public Events? Events { get; set; }

        [JsonProperty("urls")]
        public List<Url>? Urls { get; set; }

        public bool Favorite { get; set; }
    }

    public class Comics
    {
        [JsonProperty("available")]
        public int Available { get; set; }

        [JsonProperty("collectionURI")]
        public string? CollectionURI { get; set; }

        [JsonProperty("items")]
        public List<Item>? Items { get; set; }

        [JsonProperty("returned")]
        public int Returned { get; set; }
    }

    public class Events
    {
        [JsonProperty("available")]
        public int Available { get; set; }

        [JsonProperty("collectionURI")]
        public string? CollectionURI { get; set; }

        [JsonProperty("items")]
        public List<Item>? Items { get; set; }

        [JsonProperty("returned")]
        public int Returned { get; set; }
    }

    public class Item
    {
        [JsonProperty("resourceURI")]
        public string? ResourceURI { get; set; }

        [JsonProperty("name")]
        public string? Name { get; set; }

        [JsonProperty("type")]
        public string? Type { get; set; }
    }

    public class Series
    {
        [JsonProperty("available")]
        public int Available { get; set; }

        [JsonProperty("collectionURI")]
        public string? CollectionURI { get; set; }

        [JsonProperty("items")]
        public List<Item>? Items { get; set; }

        [JsonProperty("returned")]
        public int Returned { get; set; }
    }

    public class Stories
    {
        [JsonProperty("available")]
        public int Available { get; set; }

        [JsonProperty("collectionURI")]
        public string? CollectionURI { get; set; }

        [JsonProperty("items")]
        public List<Item>? Items { get; set; }

        [JsonProperty("returned")]
        public int Returned { get; set; }
    }

    public class Thumbnail
    {
        [JsonProperty("path")]
        public string? Path { get; set; }

        [JsonProperty("extension")]
        public string? Extension { get; set; }
    }

    public class Url
    {
        [JsonProperty("type")]
        public string? Type { get; set; }

        [JsonProperty("url")]
        public string? UrlAcess { get; set; }
    }
}

File: app/lib/models/Character.dart

class Character {
  int? id;
  String? name;
  String? description;
  String? modified;
  Thumbnail? thumbnail;
  bool? favorite;
  List<Thumbnail>? thumbnailComics;

  Character(
      {this.id,
        this.name,
        this.description,
        this.modified,
        this.thumbnail,
        this.favorite,
        this.thumbnailComics});

  Character.fromJson(Map<String, dynamic> json) {
    id = json['id'];
    name = json['name'];
    description = json['description'];
    modified = json['modified'];
    thumbnail = json['thumbnail'] != null
        ? Thumbnail.fromJson(json['thumbnail'])
        : null;
    favorite = json['favorite'];
    if (json['thumbnailComics'] != null) {
      thumbnailComics = <Thumbnail>[];
      json['thumbnailComics'].forEach((v) {
        thumbnailComics!.add(Thumbnail.fromJson(v));
      });
    }
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = <String, dynamic>{};
    data['id'] = id;
    data['name'] = name;
    data['description'] = description;
    data['modified'] = modified;
    if (thumbnail != null) {
      data['thumbnail'] = thumbnail!.toJson();
    }
    data['favorite'] = favorite;
    if (thumbnailComics != null) {
      data['thumbnailComics'] =
          thumbnailComics!.map((v) => v.toJson()).toList();
    }
    return data;
  }
}

class Thumbnail {
  String? path;
  String? extension;

  Thumbnail({this.path, this.extension});

  Thumbnail.fromJson(Map<String, dynamic> json) {
    path = json['path'];
    extension = json['extension'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = <String, dynamic>{};
    data['path'] = path;
    data['extension'] = extension;
    return data;
  }

  String Url() {
    return "${path!}.${extension!}";
  }
}
